TTS/tts/models/xtts.py

```diff
@@ -354,12 +374,38 @@ def get_gpt_cond_latents(self, audio, sr, length: int = 3):
 
         Args:
             audio_path (str): Path to the audio file.
+            sr (int): Sample rate of the audio.
             length (int): Length of the audio in seconds. Defaults to 3.
         """
 
         audio_22k = torchaudio.functional.resample(audio, sr, 22050)
         audio_22k = audio_22k[:, : 22050 * length]
-        mel = wav_to_mel_cloning(audio_22k, mel_norms=self.mel_stats.cpu())
+        if self.args.gpt_use_perceiver_resampler:
+            mel = wav_to_mel_cloning(audio_22k,
+                mel_norms=self.mel_stats.cpu(),
+                n_fft=2048,
+                hop_length=256,
+                win_length=1024,
+                power=2,
+                normalized=False,
+                sample_rate=22050,
+                f_min=0,
+                f_max=8000,
+                n_mels=80
+            )
+        else:
+            mel = wav_to_mel_cloning(audio_22k,
+                mel_norms=self.mel_stats.cpu(),
+                n_fft=4096,
+                hop_length=1024,
+                win_length=4096,
+                power=2,
+                normalized=False,
+                sample_rate=22050,
+                f_min=0,
+                f_max=8000,
+                n_mels=80
+            )
```

Comment:
This segment would be easier to read if it was less copy-pasted – now you need to read through all of it to find out that only three arguments were changed.

Reply:
I agree. I'll deal with it.